File: frontend/src/app/model/order.ts
export interface Order {
    id: number;
    buyerId: string;
    shippingAddress: ShippingAddress;
    orderData: string;
    orderItems: OrderItem[];
    subtotal: number;
    deliveryFee: number;
    orderStatus: string;
    total: number;
}

export interface OrderItem {
    productId: number;
    name: string;
    pictureUrl: string;
    price: number;
    quantity: number;
}

export interface ShippingAddress {
    fullName: string;
    address1: string;
    address2: string;
    city: string;
    state: string;
    zip: string;
    country: string;
}

File: Backend/Domain/Entity/OrderAggregate/Order.cs
namespace Backend.Domain.Entity.OrderAggregate;

public class Order
{
    public int Id { get; set; }
    public required string BuyerId { get; set; }
    public required ShippingAddress ShippingAddress { get; set; }
    public DateTime OrderData { get; set; } = DateTime.UtcNow;
    public List<OrderItem> OrderItems { get; set; } = [];
    public long Subtotal { get; set; }
    public long DeliveryFee { get; set; }
    public OrderStatus OrderStatus { get; set; } = OrderStatus.Pending;
    public required string PaymentIntentId { get; set; }

    public long GetTotal() => Subtotal + DeliveryFee;
}